displayLinac.py

```diff
@@ -120,7 +120,18 @@ def create_faults(self):
                 pv = self.pv_addr(suffix)
 
             elif level == "CM":
-                pv = self.cryomodule.pv_addr(suffix)
+                cm_type = csvFaultDict["CM Type"]
+                prefix = csvFaultDict["PV Prefix"].format(
+                    LINAC=self.linac.name,
+                    CRYOMODULE=self.cryomodule.name,
+                    CAVITY=self.number
+                )
+
+                if (cm_type == "1.3" and self.cryomodule.is_harmonic_linearizer) or (
+                        cm_type == "3.9" and not self.cryomodule.is_harmonic_linearizer
+                ):
+                    continue
```

Comment:
What is this if statement doing? Is this a debug statement trying to catch an impossible scenario?

Reply:
This is the chunk of code that ignores the fault if it doesn't apply to that CM. So for magnets, since it only applies to 1.3s, it skips creating the fault if the CM is a 3.9

Reply:
I forgot who 1.3s and 3.9s apply to. So is 1.3 the normal cryomodule frequency? And 3.9 the harmonic linearizer?

Reply:
Yup!

Reply:
The code definitely works per testing, but I realize I'm missing something here lol. Is the first chunk of the if statement looking for something that's 1.3 (aka normal cm) AND is a harmonic linearizer?